### PR TITLE
Adding security section.

### PR DESCRIPTION
**Pull request by jmighion**
I tried my best at the adoc parts. Let me know if there's any formatting requirements that I may have missed, or anything in general.

This is for https://issues.redhat.com/browse/AAP-17962

### CODE REVIEW

File: stories/topics/con-azure-security.adoc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+The following are a few topics to highlight the security posture of the application and the infrastructure.
+
+* Data encryption in flight and at rest
+** link:https://learn.microsoft.com/en-us/azure/security/fundamentals/encryption-atrest#azure-storage[All Azure Storage Services enable server-side encryption by default using service-managed keys]
+** link:https://learn.microsoft.com/en-us/azure/security/fundamentals/encryption-atrest#conclusion[All Azure hosted services are committed to providing Encryption at Rest options]
+** link:https://learn.microsoft.com/en-us/azure/security/fundamentals/encryption-overview[Azure encryption overview]
+** All communications between services within AKS (AAP, Postgres, storage accounts, etc) use TLS v1.2 or higher.
+** link:https://learn.microsoft.com/en-us/security/benchmark/azure/baselines/azure-kubernetes-service-aks-security-baseline[Azure security baseline for Azure Kubernetes Service (AKS)]
+* Password storage
+** Customer supplied AAP admin password is encrypted in transit.
```
Comment (ariordan-redhat) [inline stories/topics/con-azure-security.adoc:15]:
```suggestion
** The customer-supplied {PlatformNameShort} admin password is encrypted in transit.
```

File: stories/topics/con-azure-security.adoc [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+= Security
+
+{AAPonAzureNameShort} follows security best practices from both the Red Hat and Microsoft product.
+The following are a few topics to highlight the security posture of the application and the infrastructure.
+
+* Data encryption in flight and at rest
+** link:https://learn.microsoft.com/en-us/azure/security/fundamentals/encryption-atrest#azure-storage[All Azure Storage Services enable server-side encryption by default using service-managed keys]
+** link:https://learn.microsoft.com/en-us/azure/security/fundamentals/encryption-atrest#conclusion[All Azure hosted services are committed to providing Encryption at Rest options]
+** link:https://learn.microsoft.com/en-us/azure/security/fundamentals/encryption-overview[Azure encryption overview]
+** All communications between services within AKS (AAP, Postgres, storage accounts, etc) use TLS v1.2 or higher.
```
Comment (ariordan-redhat) [inline stories/topics/con-azure-security.adoc:12]:
```suggestion
** All communications between services within AKS (for example, {PlatformNameShort}, Postgres, storage accounts) use TLS v1.2 or higher.
```

File: stories/topics/con-azure-security.adoc [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+** link:https://learn.microsoft.com/en-us/security/benchmark/azure/baselines/azure-kubernetes-service-aks-security-baseline[Azure security baseline for Azure Kubernetes Service (AKS)]
+* Password storage
+** Customer supplied AAP admin password is encrypted in transit.
+It is accessible to SREs from the kubernetes API and can be reset by the SREs upon customer request.
+* Keys generated with industry standards
+** link:https://learn.microsoft.com/en-us/azure/security/fundamentals/encryption-overview#azure-storage-service-encryption[Data in Azure Storage is encrypted and decrypted transparently using 256-bit AES encryption, one of the strongest block ciphers available, and is FIPS 140-2 compliant]
+* Key installation, rotation
+** link:https://learn.microsoft.com/en-us/azure/security/fundamentals/encryption-models#key-access[Microsoft manages key rotation, backup, and redundancy]
+* SSL/TLS traffic encryption
+** All communications between services within AKS (AAP, Postgres, storage accounts, etc) use TLS v1.2 or higher.
```
Comment (ariordan-redhat) [inline stories/topics/con-azure-security.adoc:22]:
```suggestion
** All communications between services within AKS (for example, {PlatformNameShort}, Postgres, storage accounts) use TLS v1.2 or higher.
```
Does this need to be added twice (in both the data encryption items and SSL?TLS)?

File: stories/topics/con-azure-security.adoc [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+** All communications to AAP UIs either via the application gateway for public deployments or the nginx ingress for private deployments use TLS v1.2 or higher.
```
Comment (ariordan-redhat) [inline stories/topics/con-azure-security.adoc:23]:
```suggestion
** All communications to {PlatformNameShort} UIs, either via the application gateway for public deployments or the nginx ingress for private deployments, use TLS v1.2 or higher.
```

File: stories/topics/con-azure-security.adoc [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+* API security
+** Any parts of the AAP APIs that could leak any sensitive information are only accessible via authenticating as a known AAP user and requires that user to have the right level of authorization to use those APIs.
```
Comment (ariordan-redhat) [inline stories/topics/con-azure-security.adoc:25]:
```suggestion
** Any parts of the {PlatformNameShort} APIs that could leak any sensitive information are only accessible via authenticating as a known {PlatformNameShort} user and require that user to have the right level of authorization to use those APIs.
```

File: stories/topics/con-azure-security.adoc [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+In a private deployment, access to the AAP APIs is only accessible to the customer via the route they choose to connect to the private deployment.
```
Comment (ariordan-redhat) [inline stories/topics/con-azure-security.adoc:26]:
```suggestion
In a private deployment, access to the {PlatformNameShort} APIs is only accessible to the customer via the route they choose to connect to the private deployment.
```

File: stories/topics/con-azure-security.adoc [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+SREs have access to the customer AAP though because we are managing it for the customers.
```
Comment (ariordan-redhat) [inline stories/topics/con-azure-security.adoc:27]:
```suggestion
However, Red Hat SREs have access a shared and secure managed application resource group on customer {AAPonAzureNameShort} applications to perform management operations.
```

File: stories/topics/con-azure-security.adoc [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+** The kubernetes API is private and only accessible from a private endpoint
```
Comment (ariordan-redhat) [inline stories/topics/con-azure-security.adoc:28]:
```suggestion
** The Kubernetes API is private and only accessible from a private endpoint
```

File: stories/topics/con-azure-security.adoc [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+** Workload identity is enabled and it allows Kubernetes applications to access Azure cloud resources securely with Azure AD.
+* Updates and patching
+** The Red Hat SREs regularly update both the kubernetes version, underlying node OS, and AAP version to the latest available stable versions to get the latest features, bug fixes, and security fixes.
```
Comment (ariordan-redhat) [inline stories/topics/con-azure-security.adoc:31]:
```suggestion
** The Red Hat SREs regularly update the Kubernetes version, underlying node OS, and {PlatformNameShort} version to the latest available stable versions to get the latest features, bug fixes, and security fixes.
```

File: stories/topics/con-azure-security.adoc (hunk 7 above)
Comment (jmighion) [reply stories/topics/con-azure-security.adoc]: Removed this entirely.

**Comment by jmighion** [comment]
Thanks for the review!

**Review by ariordan-redhat — APPROVED** [review]
LGTM

**Comment by ariordan-redhat** [comment]
@jmighion Do you want this merged / published straight away?

**Comment by jmighion** [comment]
@ariordan-redhat Yep! Whenever you can.

**Pull request merged**